### PR TITLE
Refreshed docs for v5.0.0

### PR DESCRIPTION
**Pull request by psrok1**
TODO:

- [x] How to extend configuration in own Karton services?
- [x] Documentation of configuration fields and arguments
- [x] How to derive from ready-made Karton services?

### CODE REVIEW

File: docs/what_new.rst [hunk 1]
```diff
@@ -3,6 +3,39 @@ Breaking changes
...
+
+  .. code-block:: python
+
+    if __name__ == "__main__":
+        MyConsumer.main()  # correct
+
+    if __name__ == "__main__":
+        MyConsumer().main()  # throws TypeError
+
+* ``karton.core.Consumer.process`` no longer accepts no arguments. First argument of this method is incoming task.
```
Comment (nazywam) [inline docs/what_new.rst:24]:
```suggestion
* ``karton.core.Consumer.process`` no longer accepts no arguments. First argument of this method is incoming the task.
```

**Review by nazywam — COMMENTED** [review]
Good so far

File: docs/index.rst [hunk 2]
```diff
@@ -14,7 +14,8 @@ Karton ecosystem consists of:
 
 - `Redis <https://www.redis.io/>`_ - store used for message exchange between Karton subsystems
 
-- `MinIO <https://github.com/minio/minio>`_ - temporary object storage compatible with Amazon S3 API, holds all the heavy objects (aka Resources) like samples, analyses or dumps.
+- Temporary object storage compatible with Amazon S3 API, holds all the heavy objects (aka Resources) like samples, analyses or dumps.
```
Comment (nazywam) [inline docs/index.rst:17]:
```suggestion
- Temporary object storage compatible with Amazon S3 API, holds all the heavy objects (aka Resources) like samples, analyses or memory dumps.
```

File: docs/what_new.rst [hunk 3]
```diff
@@ -3,6 +3,61 @@ Breaking changes
 
 This chapter will describe significant changes introduced in major version releases of Karton. Versions before 4.0.0 were not officially released, so they have value only for internal purposes. Don't worry about it if you are a new user.
 
+What changed in Karton 5.0.0
+----------------------------
+
+Karton-System and core services are still able to communicate with previous versions.
+
+* Changed ``karton.ini`` section name containing S3 client configuration from ``[minio]`` to ``[s3]``.
```
Comment (nazywam) [inline docs/what_new.rst:11]:
```suggestion
* Changed name of ``karton.ini`` section that contains S3 client configuration from ``[minio]`` to ``[s3]``.
```

File: docs/what_new.rst [hunk 4]
```diff
@@ -3,6 +3,61 @@ Breaking changes
...
+
+  In addition to this, you need to add URI scheme to the ``address`` field and remove the ``secure`` field.
```
Comment (nazywam) [inline docs/what_new.rst:13]:
```suggestion
  In addition to this, you need to add a URI scheme to the ``address`` field and remove the ``secure`` field.
```

File: docs/what_new.rst [hunk 5]
```diff
@@ -3,6 +3,61 @@ Breaking changes
...
+    - address = localhost:9000
+    + address = http://localhost:9000
+      bucket = karton
+    - secure = 0
+
+  v5.0.0 maps ``[minio]`` configuration to correct ``[s3]`` configuration internally, but ``[minio]`` scheme
+  is considered deprecated and can be removed in further major release.
+
+* Karton library uses `Boto3 <https://github.com/boto/boto3>`_ library as a S3 client instead of `Minio-Py <https://github.com/minio/minio-py>`_ underneath.
+  It may be important to check if your code relies on exceptions thrown by previous S3 client.
```
Comment (nazywam) [inline docs/what_new.rst:31]:
```suggestion
  You may want to check if your code relies on exceptions thrown by previous S3 client.
```

File: docs/breaking_changes.rst [hunk 6]
```diff
@@ -3,8 +3,63 @@ Breaking changes
...
+
+  .. code-block:: python
+
+    if __name__ == "__main__":
+        MyConsumer.main()  # correct
+
+    if __name__ == "__main__":
+        MyConsumer().main()  # throws TypeError
+
+* ``karton.core.Consumer.process`` no longer accepts no arguments. First argument of this method is incoming the task.
```
Comment (nazywam) [inline docs/breaking_changes.rst:46]:
```suggestion
* ``karton.core.Consumer.process`` no longer accepts no arguments. First argument of this method is the incoming task.
```

File: docs/getting_started.rst [hunk 7]
```diff
@@ -22,16 +22,15 @@ Each Karton subsystem needs a ``karton.ini`` file that contains the connection p
 
 You can also use this file to store custom fields and use them e.g. by :ref:`example-overriding-config`.
```
Comment (nazywam) [inline docs/getting_started.rst:23]: image

CI seems angry about this line :thinking:

File: docs/getting_started.rst [hunk 8]
```diff
@@ -127,6 +132,7 @@ If we now run the consumer and spawn a few "foobar" tasks we should get a few fo
     Bar
     [INFO] Task done - d3a39940-d64c-4033-a7da-80eae9786631
 
+Check :ref:`Karton service examples` for more details.
```
Comment (nazywam) [inline docs/getting_started.rst:135]: same here 
image

File: docs/service_configuration.rst [hunk 9]
```diff
@@ -0,0 +1,189 @@
...
+ [s3]           secret_key        S3 API secret key (password)
+ [s3]           bucket            Default bucket name for storing produced resources
+ [redis]        host              Redis server hostname
+ [redis]        port              Redis server port
+ [redis]        db                Redis server database id (default: 0)
+ [redis]        username          Redis server AUTH username (default: None)
+ [redis]        password          Redis server AUTH password (default: None)
+ [redis]        socket_timeout    Socket timeout for Redis operations in seconds (default: 30, use 0 to turn off if timeout doesn't work properly)
+ [karton]       identity          Karton service identity override (overrides the name provided in class / constructor arguments)
+ [karton]       persistent        Karton service persistency override
```
Comment (nazywam) [inline docs/service_configuration.rst:50]:
```suggestion
 [karton]       persistent        Karton service queue persistency override
```

File: docs/service_configuration.rst [hunk 10]
```diff
@@ -0,0 +1,189 @@
...
+ [karton]       task_timeout      Karton service task execution timeout in seconds. If your service sometimes hangs, Karton will schedule SIGALRM if this value is set.
```
Comment (nazywam) [inline docs/service_configuration.rst:51]:
```suggestion
 [karton]       task_timeout      Karton service task execution timeout in seconds. Useful if your service sometimes hangs. Karton will schedule SIGALRM if this value is set.
```

File: docs/service_configuration.rst [hunk 11]
```diff
@@ -0,0 +1,189 @@
...
+ [logging]      level             Logging level for Karton service logger (default: INFO)
+ [signaling]    status            Turns on producing of 'karton.signaling.status' tasks, signalling the task start and finish events by Karton service (default: 0, off)
+============   ===============   =======================================================================================================================================
+
+
+Karton System configuration
+---------------------------
+
+Most core services can be tuned up depending on your needs. Custom service configuration is handled the same way as general Karton configuration.
```
Comment (nazywam) [inline docs/service_configuration.rst:60]:
```suggestion
Most core services can be tuned depending on your needs. Custom service configuration is handled the same way as general Karton configuration.
```

File: docs/service_configuration.rst [hunk 12]
```diff
@@ -0,0 +1,189 @@
...
+Good example is Karton System:
+
+============   =========================   =======================================================================================================================================
+ Section        Option                       Description
+============   =========================   =======================================================================================================================================
+ [system]       gc_interval                  Spawn interval for garbage collection tasks in seconds. Default is 3 minutes.
+ [system]       task_dispatched_timeout      Timeout for tasks that are stuck in DISPATCHED state (e.g. Producer crashed during upload of resources). Default is 24 hours.
+ [system]       task_started_timeout         Timeout for tasks that are stuck in STARTED state (e.g. non-graceful crash of Consumer during task processing). Default is 24 hours.
+ [system]       task_crashed_timeout         Timeout for removal of crashed tasks. Default is 3 days.
+ [system]       enable_gc                    Enable garbage collection. GC can be turned off if you want to scale up routing via few Karton System instances.
```
Comment (nazywam) [inline docs/service_configuration.rst:71]:
```suggestion
 [system]       enable_gc                    Enable garbage collection. GC can be turned off if you want to scale up routing using several Karton System instances.
```

File: docs/service_configuration.rst [hunk 13]
```diff
@@ -0,0 +1,189 @@
...
+Extending configuration
+-----------------------
+
+During development of your own Karton services you may want to provide your own configuration fields.
+
+All configuration set in ``karton.ini`` files and ``KARTON_`` envs is available in ``self.config`` object and doesn't
+require additional definition.
+
+The only thing that needs to be extended is argument parser if you want to use command-line arguments. Fortunately,
+Karton classes expose dedicated methods for that purpose.
```
Comment (nazywam) [inline docs/service_configuration.rst:114]:
```suggestion
Karton classes expose dedicated methods for this purpose.
```

Not super sure tho?

File: docs/service_configuration.rst [hunk 14]
```diff
@@ -0,0 +1,189 @@
...
+                            Timeout for non-enqueued tasks stuck in Started state (non-graceful shutdown of consumer)
+      --task-crashed-timeout TASK_CRASHED_TIMEOUT
+                            Timeout for tasks in Crashed state
+
+Extending configuration
+-----------------------
+
+During development of your own Karton services you may want to provide your own configuration fields.
+
+All configuration set in ``karton.ini`` files and ``KARTON_`` envs is available in ``self.config`` object and doesn't
```
Comment (nazywam) [inline docs/service_configuration.rst:110]:
```suggestion
All configuration values set in ``karton.ini`` files and ``KARTON_`` envs are available in ``self.config`` object and don't
```

File: docs/service_configuration.rst [hunk 15]
```diff
@@ -0,0 +1,189 @@
...
+        SmolKarton.main()
+
+``args_parser`` method exposes the ``argparse.ArgumentParser`` that is used for handling CLI arguments. Values from
+argparse are then passed to ``config_from_args`` that maps arguments into sections and options of configuration.
+That mechanism allows you to define your own arguments and include these values in the final configuration.
+
+Customizing ready-made Karton services
+--------------------------------------
+
+Ready-made Karton services like ``karton-mwdb-reporter`` are coming with predefined set of filters and emitted headers.
```
Comment (nazywam) [inline docs/service_configuration.rst:170]:
```suggestion
Ready-made Karton services like ``karton-mwdb-reporter`` are coming with a predefined set of filters and emitted headers.
```

**Review by nazywam — APPROVED** [review]

File: docs/getting_started.rst (hunk 7 above)
Comment (psrok1) [reply docs/getting_started.rst]: Fixed

**Pull request merged**